TaskManager.Data/Unitofwork.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace TaskManager.Data
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.EntityFrameworkCore;
+    using TaskManager.Data.Entities;
+    using TaskManager.Data.Repositories;
+    using TaskManager.Models;
+
+    public class UnitOfWork : IDisposable, IUnitOfWork
+
+    {
+        private readonly TaskContext taskContext;
+        private TaskRepository taskRepository;
+
+        public UnitOfWork(TaskContext taskContext)
+        {
+            this.taskContext = taskContext;
```

Comment:
this.taskContext = taskContext ?? throw new ArgumentNullException(nameof(taskContext));